packages/typespec-vscode/src/npm-utils.ts

```diff
@@ -0,0 +1,150 @@
+import { readFile } from "fs/promises";
+import path from "path";
+import semver from "semver";
+import logger from "./log/logger.js";
+import { ExecOutput, loadModule, spawnExecution, spawnExecutionEvents } from "./utils.js";
+
+export enum InstallationAction {
+  Install = "Install",
+  Upgrade = "Upgrade",
+  Skip = "Skip",
+  Cancel = "Cancel",
+}
+
+export enum npmDependencyType {
+  dependencies = "dependencies",
+  peerDependencies = "peerDependencies",
+  devDependencies = "devDependencies",
+}
+
+export interface NpmPackageInfo {
```

Comment:
there is a NodePackage type in compiler you can use directly

Reply:
VSCode Extension may not library dependent on `@typespec/compiler`. So we may not import type from compiler.

Reply:
We can use the type defined in compiler by import type {...}